app/validators/key_uniqueness_validator.rb

```diff
@@ -0,0 +1,14 @@
+class KeyUniquenessValidator < ActiveModel::Validator
+  def validate(record)
+    key = options[:register_name]
+    records = options[:records]
+    is_create = options[:is_create]
+    if ActiveModel::Type::Boolean.new.cast(is_create)
+      register_sym = key.underscore.to_sym
+      key_param = record.instance_variable_get("@#{key.underscore}")
+      if records.select {|r| r[:item][key] == key_param}.present?
```

Comment:
Note, this is a lookup from the in-memory RSF, so currently will not be updated with live changes. We could use the old style record lookup but that would require an additional HTTP call.